packages/runtime/datastore/src/test/dataStoreLayerCompatValidation.spec.ts

```diff
@@ -0,0 +1,284 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import type {
+	ILayerCompatDetails,
+	ILayerCompatSupportRequirements,
+} from "@fluid-internal/client-utils";
+import {
+	FluidErrorTypes,
+	type ITelemetryBaseProperties,
+} from "@fluidframework/core-interfaces/internal";
+import type { IChannel } from "@fluidframework/datastore-definitions/internal";
+import { UsageError } from "@fluidframework/telemetry-utils/internal";
+import { MockFluidDataStoreContext } from "@fluidframework/test-runtime-utils/internal";
+import Sinon from "sinon";
+
+import {
+	DataStoreCompatDetailsForRuntime,
+	RuntimeSupportRequirements,
+	validateRuntimeCompatibility,
+} from "../dataStoreLayerCompatState.js";
+import { FluidDataStoreRuntime, ISharedObjectRegistry } from "../dataStoreRuntime.js";
+import { pkgVersion } from "../packageVersion.js";
+
+type ILayerCompatSupportRequirementsOverride = Omit<
```

Comment:
If I ever merge #23694 you could use `Mutable` here like `= Mutable<ILayerCompatSupportRequirements, "requiredFeatures">`